network-config-analyzer/CalicoPolicyYamlParser.py

```diff
@@ -624,7 +623,14 @@ def parse_policy(self):
 
         policy_spec = self.policy['spec']
         self._set_affects_ingress_egress(policy_spec, is_profile, res_policy)
-        self._set_selected_peers(policy_spec, is_profile, res_policy)
+        res_policy.selected_peers = self._set_selected_peers(policy_spec, is_profile, res_policy.name)
+        ns_selector = policy_spec.get('namespaceSelector')
+        if ns_selector:
+            if not is_global_np:
+                self.syntax_error(f'namespaceSelector is not allowed in the spec of Calico {res_policy.policy_kind}')
```

Comment:
if you update the function `_set_affects_ingress_egress ` as in my previous comment, we can remove this check here